app/models/service_provider.rb

```diff
@@ -147,6 +148,15 @@ def null_certificate
   end
   # rubocop:enable Rails/TimeZone
 
+  def logo_is_less_than_mb
+    return unless logo_file.attached?
+
+    if logo_file.blob.byte_size > 1.megabytes
+      errors.add(:logo_file, 'Logo must be less than 1MB')
+      logo_file = nil # rubocop:disable Lint/UselessAssignmentl
```

Comment:
can't use `logo_file.purge` like i'd like to because the model doesn't exist yet, so this throws an error :(